File: guests/swift/progress.swift
// The progress scene, Swift port — guests/rust/progress.rs,
// tools/scenes/progress.steps.

import Foundation
import Kaya

KayaApp.run { app in
    app.build { tx in
        tx.window(title: "progress")
        let root = tx.column { root in
            tx.progress(value: 0.25)  // progress#0
            tx.progress(indeterminate: true)  // progress#1
            return root
        }
        tx.mount(root)
    }
}

File: guests/ocaml/progress.ml
(* The progress scene, OCaml port — guests/rust/progress.rs,
   tools/scenes/progress.steps. *)

open Kaya_app

let () =
  let app = Kaya_app.create () in

  build app (fun () ->
     window ~title:"progress" ();
     let root =
       column
         [
           progress ~value:0.25 (* progress#0 *);
           progress ~indeterminate:true (* progress#1 *);
         ]
         ()
     in
     mount root);

  exit (run app)
